1.3.48 will error out on private YouTube videos

I am a Lavalink user using Lavaplayer 1.3.48. On trying to load a private YouTube video, the following error is shown:

```
2020-05-15 19:59:34.279  INFO 12408 --- [  XNIO-1 task-7] l.server.player.AudioLoaderRestHandler   : GET /loadtracks
2020-05-15 19:59:34.731  INFO 12408 --- [ader-2-thread-1] lavalink.server.player.AudioLoader       : Loaded track null
2020-05-15 19:59:34.735 ERROR 12408 --- [  XNIO-1 task-8] io.undertow.request                      : UT005023: Exception handling request to /loadtracks

org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1013) ~[spring-webmvc-5.1.9.RELEASE.jar!/:5.1.9.RELEASE]
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:897) ~[spring-webmvc-5.1.9.RELEASE.jar!/:5.1.9.RELEASE]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:686) ~[jboss-servlet-api_4.0_spec-1.0.0.Final.jar!/:1.0.0.Final]
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:882) ~[spring-webmvc-5.1.9.RELEASE.jar!/:5.1.9.RELEASE]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:791) ~[jboss-servlet-api_4.0_spec-1.0.0.Final.jar!/:1.0.0.Final]
        at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:81) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchToPath(ServletInitialHandler.java:186) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.spec.AsyncContextImpl$2$1.handleRequest(AsyncContextImpl.java:194) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:376) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.spec.AsyncContextImpl$2.run(AsyncContextImpl.java:191) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.spec.AsyncContextImpl$6.run(AsyncContextImpl.java:465) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.servlet.spec.AsyncContextImpl$TaskDispatchRunnable.run(AsyncContextImpl.java:584) ~[undertow-servlet-2.0.26.Final.jar!/:2.0.26.Final]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
Caused by: java.lang.NullPointerException: null
        at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:347) ~[na:na]
        at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:323) ~[na:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.encodeTrack(DefaultAudioPlayerManager.java:249) ~[lavaplayer-1.3.48.jar!/:na]
        at lavalink.server.util.Util.toMessage(Util.java:49) ~[classes!/:na]
        at lavalink.server.player.AudioLoaderRestHandler.lambda$encodeLoadResult$0(AudioLoaderRestHandler.java:88) ~[classes!/:na]
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1540) ~[na:na]
        at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085) ~[na:na]
        at lavalink.server.player.AudioLoaderRestHandler.encodeLoadResult(AudioLoaderRestHandler.java:83) ~[classes!/:na]
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642) ~[na:na]
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[na:na]
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[na:na]
        at lavalink.server.player.AudioLoader.trackLoaded(AudioLoader.java:72) ~[classes!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItemOnce(DefaultAudioPlayerManager.java:443) ~[lavaplayer-1.3.48.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItem(DefaultAudioPlayerManager.java:419) ~[lavaplayer-1.3.48.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$createItemLoader$0(DefaultAudioPlayerManager.java:218) ~[lavaplayer-1.3.48.jar!/:na]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
        ... 3 common frames omitted
```

I also received the following from Lavalink, in case it helps debugging.

```
Lavalink API returned an unsupported response, Please report it.
Query: http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//www.youtube.com/watch%3Fv%3DY2zKSExZe5E
{'timestamp': '2020-05-15T17:52:39.230+0000', 'status': 500, 'error': 'Internal Server Error', 'message': 'Internal Server Error', 'path': '/loadtracks', 'query': 'https://www.youtube.com/watch?v=Y2zKSExZe5E', 'encodedquery': 'http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//www.youtube.com/watch%3Fv%3DY2zKSExZe5E', 'loadType': <LoadType.LOAD_FAILED: 'LOAD_FAILED'>}
Query: http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//www.youtube.com/watch%3Fv%3DY2zKSExZe5E
'timestamp': '2020-05-15T17:53:55.541+0000', 'status': 500, 'error': 'Internal Server Error', 'message': 'Internal Server Error', 'path': '/loadtracks', 'query': 'https://www.youtube.com/watch?v=Y2zKSExZe5E', 'encodedquery': 'http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//www.youtube.com/watch%3Fv%3DY2zKSExZe5E', 'loadType': <LoadType.LOAD_FAILED: 'LOAD_FAILED'>}
Query: http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//youtu.be/GuTQdDI_9DE
{'timestamp': '2020-05-15T17:59:34.742+0000', 'status': 500, 'error': 'Internal Server Error', 'message': 'Internal Server Error', 'path': '/loadtracks', 'query': 'https://youtu.be/GuTQdDI_9DE', 'encodedquery': 'http://[SERVER IP HERE]:2333/loadtracks?identifier=https%3A//youtu.be/GuTQdDI_9DE', 'loadType': <LoadType.LOAD_FAILED: 'LOAD_FAILED'>}
```

Tracks to test with:
https://www.youtube.com/watch?v=GuTQdDI_9DE
https://www.youtube.com/watch?v=Y2zKSExZe5E

--- a/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
+++ b/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
@@ -102,6 +102,12 @@ public class DefaultYoutubeTrackDetailsLoader implements YoutubeTrackDetailsLoad
       String unplayableReason = getUnplayableReason(statusBlock);
       throw new FriendlyException(unplayableReason, COMMON, null);
     } else if ("LOGIN_REQUIRED".equals(status)) {
+      String errorReason = statusBlock.get("errorScreen").get("playerErrorMessageRenderer").get("reason").get("simpleText").text();
+
+      if ("Private video".equals(errorReason)) {
+        return InfoStatus.DOES_NOT_EXIST;
+      }
+
       return InfoStatus.REQUIRES_LOGIN;
     } else {
       throw new FriendlyException("This video cannot be viewed anonymously.", COMMON, null);

--- a/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
+++ b/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
@@ -109,7 +109,7 @@ public class DefaultYoutubeTrackDetailsLoader implements YoutubeTrackDetailsLoad
               .text();
 
       if ("Private video".equals(errorReason)) {
-        return InfoStatus.DOES_NOT_EXIST;
+        throw new FriendlyException("This is a private video.", COMMON, null);
       }
 
       return InfoStatus.REQUIRES_LOGIN;

--- a/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
+++ b/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/DefaultYoutubeTrackDetailsLoader.java
@@ -102,7 +102,11 @@ public class DefaultYoutubeTrackDetailsLoader implements YoutubeTrackDetailsLoad
       String unplayableReason = getUnplayableReason(statusBlock);
       throw new FriendlyException(unplayableReason, COMMON, null);
     } else if ("LOGIN_REQUIRED".equals(status)) {
-      String errorReason = statusBlock.get("errorScreen").get("playerErrorMessageRenderer").get("reason").get("simpleText").text();
+      String errorReason = statusBlock.get("errorScreen")
+              .get("playerErrorMessageRenderer")
+              .get("reason")
+              .get("simpleText")
+              .text();
 
       if ("Private video".equals(errorReason)) {
         return InfoStatus.DOES_NOT_EXIST;

